toml/decoder.py

```diff
@@ -326,9 +349,50 @@ def loads(s, _dict=dict, decoder=None):
     multilinestr = ""
     multibackslash = False
     pos = 0
+    comment_norm_factor = 0
     for idx, line in enumerate(s):
         if idx > 0:
             pos += len(s[idx - 1]) + 1
+
+        if len(comment_info) > 0 and comment_info[0][0] - 1 == idx:
+            # match the comment lines with their original positions
+            # in the file and handle them appropriately
+            original_line_no = comment_lines[0] - 1
+            original_line = (original.split('\n')[original_line_no])
+            recent = 'start'
```

Comment:
Was there something wrong with the previous approach of keeping track of recent as the file was parsed line-by-line?